### ISSUE TITLE
CS0612 warning about obsolete member not suppressed when it is referenced by a ObsoleteAttribute placed on a method

### ISSUE THREAD

**Issue by stakx** [dotnet/roslyn#23015]
**Version Used**: Visual Studio 15.4.0

**Steps to Reproduce**:

1. Create a C# console app or class library project and paste the following code into a new code file:

```csharp
using System;

[Obsolete] partial class Old { }

[Obsolete("This fixture contains tests related to " + nameof(Old) + ", which is obsolete.")]
public class A
{
    public void Test()
    {
        var old = new Old();
    }
}

public class B
{
    [Obsolete("This test is related to " + nameof(Old) + ", which is obsolete.")]  // (*)
    public void Test()
    {
        var old = new Old();
    }
}
``` 

**Expected Behavior**: No compiler warnings should be emitted about `Old` being obsolete. (The two methods that reference `Old` are themselves marked `[Obsolete]`. The `[Obsolete]` attributes that themselves refer to `Old` shouldn't trigger warnings, either.)

**Actual Behavior**: One warning `CS0612: 'Old' is obsolete` will be emitted for the line marked with `(*)` above. Note that the warning is *only* triggered for the `[Obsolete]` attribute being placed on a method. When placed on a class, no warning is emitted. That doesn't seem right to me.

![obsolete-warning](https://user-images.githubusercontent.com/104481/32413453-e157f2da-c211-11e7-9f23-da69c53cc219.png)

**Comment by MaceWindu** [comment]
should this issue also cover other attributes, that target obsolete member?
```cs
[Obsolete]
    class A
    {
    }

    class SomeAttribute : Attribute
    {
        public SomeAttribute(string _)
        {
        }
    }

    class B
    {
        [Obsolete]
        [field:Obsolete]
        [Some(nameof(A))]
        [field:Some(nameof(A))]
        public string Prop { get; }

        [field: Obsolete]
        [field: Some(nameof(A))]
        public string Prop2 { get; }
    }
```
